### PR TITLE
chore(onyx-467): migrate from savedSearches to alerts

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-467]

### Description

Moves away alerts from the stitched schema (`savedSearch`, `savedSearchesConnection`) to the unstitched one (`alerts`, `alertsConnection`) 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- migrate from savedSearches to alerts - @nickskalkin 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-467]: https://artsyproduct.atlassian.net/browse/ONYX-467?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (migrate from savedSearches to alerts - @nickskalkin)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ff0ce7096050f97e0046416bd64754d87cd21891
</p>

File: src/app/Components/Artist/ArtistHeader.tests.tsx [hunk 1]
```diff
@@ -52,48 +47,6 @@ describe("ArtistHeader", () => {
       tracks.tappedVerifiedRepresentative(mockArtist as any, partner)
     )
   })
-
-  describe("alerts set", () => {
```
Comment (nickskalkin) [inline src/app/Components/Artist/ArtistHeader.tests.tsx:56]: small clean up - alerts set feature was behind the disabled feature flag and there are no plans to enable it

File: src/app/Components/Artist/ArtistArtworks/CreateSavedSearchModal.tests.tsx [hunk 2]
```diff
@@ -23,14 +23,18 @@ import {
   tracks,
 } from "./CreateSavedSearchModal"
 
-jest.mock("../../../Scenes/SavedSearchAlert/queries/getSavedSearchIdByCriteria", () => ({
-  getSavedSearchIdByCriteria: () => Promise.resolve(null),
+jest.mock("../../../Scenes/SavedSearchAlert/queries/getAlertByCriteria", () => ({
+  getAlertByCriteria: () => Promise.resolve(null),
```
Comment (MounirDhahri) [inline src/app/Components/Artist/ArtistArtworks/CreateSavedSearchModal.tests.tsx:27]: ❗ so this is why the same behaviour didn't break here, it looks like whoever wrote this before, didn't get it to work as well and ended up mocking the file and returning `null`

File: src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tsx [hunk 3]
```diff
@@ -34,14 +34,14 @@ interface SavedSearchesListProps extends SavedSearchListWrapperProps {
 }
 
 const SORT_OPTIONS: SortOption[] = [
-  { value: "CREATED_AT_DESC", text: "Recently Added" },
+  { value: "ENABLED_AT_DESC", text: "Recently Added" },
```
Comment (MounirDhahri) [inline src/app/Scenes/SavedSearchAlertsList/Components/SavedSearchesList.tsx:37]: 👏

**Review by MounirDhahri — APPROVED** [review]
Massive change! thanks for doing this Nikita! the code looks good to me and I will do further QA on my iPhone by creating a canary build from your branch

```bash
yarn setup:releases
 ./scripts/codepush/deploy-to-codepush.sh 'Canary' 'Bye Alerts Stitching' 
```

File: src/app/Scenes/SavedSearchAlert/mutations/createSavedSearchAlert.ts [hunk 4]
```diff
@@ -5,27 +5,32 @@ import { getRelayEnvironment } from "app/system/relay/defaultEnvironment"
...
             }
           }
         }
       `,
       variables: {
-        input: {
-          attributes,
-          userAlertSettings,
-        },
+        input: input as { artistIDs: string[] }, // artistIDs is required in the input type
```
Comment (mzikherman) [inline src/app/Scenes/SavedSearchAlert/mutations/createSavedSearchAlert.ts:33]: Nice, yup - wound up doing the same in Force.

**Review by mzikherman — APPROVED** [review]
Yay! 🔥

**Review by anandaroop — APPROVED** [review]
👏🏽 👏🏽 Awesome!

**Comment by nickskalkin** [comment]
@MounirDhahri thanks a lot for testing & getting this PR merged 🙌

**Pull request merged**